packages/sommelier.rb

```diff
@@ -2,90 +2,200 @@
+    # and checksums vary with each download.
+    system 'curl https://chromium.googlesource.com/chromiumos/platform2/+archive/c6f85a3472584df37fcf2d3d99474081f29256cd.tar.gz | tar mzx --warning=no-timestamp'
+    Dir.chdir ("vm_tools/sommelier") do
+    
+    # Google's sommelier expects to find virtwl.h in their kernel source includes, but we may not have
+    # set of kernel headers which match, so we just download virtwl.h and then patch the sommelier source
+    # to look for the file locally.
+    ######################### Download virtwl.h from Chromium 5.4 kernel tree ###########################################
+    url_virtwl = "https://chromium.googlesource.com/chromiumos/third_party/kernel/+/5d641a7b7b64664230d2fd2aa1e74dd792b8b7bf/include/uapi/linux/virtwl.h?format=TEXT"
+    uri_virtwl = URI.parse url_virtwl
+    #filename_virtwl = File.basename(uri_virtwl.path)
+    filename_virtwl = 'virtwl.h_base64'
+    sha256sum_virtwl = 'a8215f4946ccf30cbd61fcf2ecc4edfe6d05bffeee0bacadd910455274955446'
+    
+    if File.exist?(filename_virtwl) && Digest::SHA256.hexdigest( File.read("./#{filename_virtwl}") ) == sha256sum_virtwl
+      puts "Unpacking virtwl source code".yellow
+    else
+      puts "Downloading virtwl".yellow
+      system('curl', '-s', '-C', '-', '--insecure', '-L', '-#', url_virtwl, '-o', filename_virtwl)
+      abort 'Checksum mismatch. :/ Try again.'.lightred unless
+        Digest::SHA256.hexdigest( File.read("./#{filename_virtwl}") ) == sha256sum_virtwl
+      puts "virtwl archive downloaded".lightgreen
+      system 'mkdir -p build/linux'
+      system 'base64 --decode virtwl.h_base64 > build/linux/virtwl.h'
+    end    
+    
+    # lld is needed so libraries linked to system libraries (e.g. libgbm.so) can be linked against, since those are required for graphics acceleration.
+    ENV['CFLAGS'] = "-fuse-ld=lld"
+    ENV['CXXFLAGS'] = "-fuse-ld=lld"
+    
+    system "meson #{CREW_MESON_OPTIONS} -Dxwayland_path=#{CREW_PREFIX}/bin/Xwayland -Dxwayland_gl_driver_path=/usr/#{ARCH_LIB}/dri -Ddefault_library=both -Dxwayland_shm_driver=noop -Dshm_driver=noop -Dvirtwl_device=/dev/null -Dpeer_cmd_prefix=\"#{CREW_PREFIX}#{PEER_CMD_PREFIX}\" build"
+    system "meson configure build"
+    system "ninja -C build"
+    
+    Dir.chdir ("build") do
+      system 'curl "https://chromium.googlesource.com/chromiumos/containers/sommelier/+/refs/heads/master/sommelierrc?format=TEXT" | base64 --decode > sommelierrc'
+      
+      system "cat <<'EOF'> .sommelier.env
+#!/bin/bash
+export CLUTTER_BACKEND=wayland
+export DISPLAY=:0
+export GDK_BACKEND=x11
+export SCALE=0.5
+export SOMMELIER_ACCELERATORS=Super_L,<Alt>bracketleft,<Alt>bracketright
+export WAYLAND_DISPLAY=wayland-1
+export XDG_RUNTIME_DIR=/var/run/chrome
+UNAME_ARCH=$(uname -m)
+if [[ \"$UNAME_ARCH\" == 'x86_64' ]] || [[ \"$UNAME_ARCH\" == 'i686' ]]
+then
+  declare -a TEMP2
+  TEMP1=$(uname -r)
+  TEMP2=(${TEMP1//[.-]/ })
```

Comment:
The 2 lines above could be combined into 1 with:
```
TEMP2=$(uname -r | cut -d'-' -f1)
```

Reply:
Fixed

Reply:
This actually doesn't put the major and minor version into the array...